sql/schemachanges: TestColumnConversions is flaky

```
~ make stress PKG=./sql/schemachange/ TESTS=TestColumnConversions
...
--- FAIL: TestColumnConversions (1.48s)
    --- FAIL: TestColumnConversions/column_conversion_checks (1.48s)
        --- FAIL: TestColumnConversions/column_conversion_checks/TIMETZ->TIME (0.07s)
        	alter_column_type_test.go:317: a TIMETZ NULL, CREATE TABLE t (
        			i INT NOT NULL,
        			a TIMETZ NULL,
        			CONSTRAINT "primary" PRIMARY KEY (i ASC),
        			FAMILY "primary" (i, a)
        		)
        	alter_column_type_test.go:317: a TIME NULL, CREATE TABLE t (
        			i INT NOT NULL,
        			a TIME NULL,
        			CONSTRAINT "primary" PRIMARY KEY (i ASC),
        			FAMILY "primary" (i, a)
        		)
        	alter_column_type_test.go:356: unexpected value found at row 0: 0000-01-01 21:11:11 -0400 -0400 (time.Time) vs 0000-01-01 01:11:11 +0000 UTC (time.Time)
FAIL


ERROR: exit status 1

1 runs completed, 1 failures, over 2s
context canceled
```

--- a/pkg/sql/schemachange/alter_column_type.go
+++ b/pkg/sql/schemachange/alter_column_type.go
@@ -122,13 +122,18 @@ var classifiers = map[sqlbase.ColumnType_SemanticType]map[sqlbase.ColumnType_Sem
 		},
 		sqlbase.ColumnType_STRING: classifierWidth,
 	},
-	// #26078
-	// sqlbase.ColumnType_TIME: {
-	// sqlbase.ColumnType_TIMETZ: ColumnConversionTrivial.classifier(),
-	// },
-	// sqlbase.ColumnType_TIMETZ: {
-	// sqlbase.ColumnType_TIME: ColumnConversionTrivial.classifier(),
-	// },
+	// TODO(bob): It looks like TIMETZ is currently in flux and the
+	// implementation, encoding, and semantics may be changing.
+	// We'll disable this change for the moment until these
+	// issues settle:
+	// https://github.com/cockroachdb/cockroach/issues/25224
+	// https://github.com/cockroachdb/cockroach/issues/26097
+	sqlbase.ColumnType_TIME: {
+		sqlbase.ColumnType_TIMETZ: ColumnConversionDangerous.classifier(),
+	},
+	sqlbase.ColumnType_TIMETZ: {
+		sqlbase.ColumnType_TIME: ColumnConversionDangerous.classifier(),
+	},
 	sqlbase.ColumnType_TIMESTAMP: {
 		sqlbase.ColumnType_TIMESTAMPTZ: ColumnConversionTrivial.classifier(),
 	},

--- a/pkg/sql/schemachange/alter_column_type_test.go
+++ b/pkg/sql/schemachange/alter_column_type_test.go
@@ -29,7 +29,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
-	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 )
 
@@ -125,14 +124,13 @@ func TestColumnConversions(t *testing.T) {
 			{SemanticType: sqlbase.ColumnType_BYTES, Width: 19}: ColumnConversionValidate,
 			{SemanticType: sqlbase.ColumnType_BYTES, Width: 20}: ColumnConversionTrivial,
 		},
-
-		// Disabled until #26078 is resolved
-		// {SemanticType: sqlbase.ColumnType_TIME}: {
-		// {SemanticType: sqlbase.ColumnType_TIMETZ}: ColumnConversionTrivial,
-		// },
-		// {SemanticType: sqlbase.ColumnType_TIMETZ}: {
-		// {SemanticType: sqlbase.ColumnType_TIME}: ColumnConversionTrivial,
-		// },
+		// See discussion in alter_column_type.go
+		{SemanticType: sqlbase.ColumnType_TIME}: {
+			{SemanticType: sqlbase.ColumnType_TIMETZ}: ColumnConversionDangerous,
+		},
+		{SemanticType: sqlbase.ColumnType_TIMETZ}: {
+			{SemanticType: sqlbase.ColumnType_TIME}: ColumnConversionDangerous,
+		},
 
 		{SemanticType: sqlbase.ColumnType_TIMESTAMP}: {
 			{SemanticType: sqlbase.ColumnType_TIMESTAMPTZ}: ColumnConversionTrivial,
@@ -246,6 +244,7 @@ func TestColumnConversions(t *testing.T) {
 						const timeTZ = "15:04:05 -0700"
 						const noZone = "2006-01-02 15:04:05"
 						const withZone = "2006-01-02 15:04:05 -0700"
+
 						var fromFmt string
 						switch from.SemanticType {
 						case sqlbase.ColumnType_TIME:
@@ -261,11 +260,12 @@ func TestColumnConversions(t *testing.T) {
 						// Always use a non-UTC zone for this test
 						const tz = "America/New_York"
 						sqlDB.Exec(t, fmt.Sprintf("SET SESSION TIME ZONE '%s'", tz))
-						loc, err := time.LoadLocation(tz)
-						if err != nil {
-							t.Fatal(err)
-						}
-						now := timeutil.Now().In(loc).Format(fromFmt)
+
+						// Use a fixed time so we don't experience weirdness when
+						// testing TIME values if the local time would roll over
+						// a day boundary when moving between Eastern and UTC.
+						// We can re-use the format strings as test data.
+						now := fromFmt
 						insert = []interface{}{now}
 
 						switch to.SemanticType {

